IMPLEMENTATION.md

```diff
@@ -0,0 +1,119 @@
+# Tour of the Java classes
+
+The package `org.openmicroscopy.client.downloader` has the most classes
+including `Download` which carries the `main` method and various other
+methods relating to different setup and download options. Its
+`ALL_GROUPS_CONTEXT` constant is used by various classes to keep OMERO
+group context from limiting query results. The `main` method uses:
+
+1. The `org.openmicroscopy.client.downloader.options` package's
+   `OptionParser` which encodes the client application's command-line
+   options. That class uses `OptionSet` as a helper for options whose
+   value represents a subset of possible choices. An instance of
+   `OptionParser.Chosen` bears the options parsed from the command line.
+
+1. OMERO.blitz's `FindChildren` to determine which OMERO model objects
+   are targeted. The `RequestManager` is a helper class for submitting
+   requests to OMERO and ensuring that the response is of the expected
+   type. A `FileMapper` instance stores the outcome of queries to
+   understand how filesets on the server relates to their images and
+   pixels instances.
+
+1. `Download.downloadFiles` to implement the `-f` option's `binary` and
+   `companion`. This is where `FileManager` is used: for downloading
+   files from a raw file store provided by the server. (The class used
+   to have more methods.) It uses `LinkMakerPaths` for linking to the
+   repository files from the other directories.
+
+1. `Download.exportImages` to implement the `-f` option's `ome-tiff` and
+   `tiff`. It uses `LocalPixels` to fetch pixel data from a raw pixels
+   store from the server, `XmlGenerator` for populating the metadata
+   store and `LinkMakerMetadata` for linking model objects in that store
+   to note their relationships. `LocalPixels` uses a couple of helpers
+   from the package `org.openmicroscopy.client.downloader.util`:
+   `TileIterator` covers all the tiles of an image and `FileIO` reads
+   and writes integers and byte arrays. Tiles are recorded in an
+   intermediate file before Bio-Formats performs the export.
+
+1. `Download.writeXmlObjects` to implement the `-f` option's
+   `ome-xml-parts` with the help of `XmlGenerator`. This also makes some
+   use of `LinkMakerPaths` for linking objects from their containers.
+   `MetadataBase` provides helpers for the other classes of the
```

Comment:
Since this introduces a new package perhaps make it clearer, e.g. `metadata.MetadataBase`, or rearrange the sentence to mention the package name first?